plugins/events-correlation-engine/src/main/java/org/opensearch/plugin/correlation/core/index/codec/CorrelationCodecVersion.java

```diff
@@ -24,15 +24,15 @@
  * @opensearch.internal
  */
 public enum CorrelationCodecVersion {
-    V_9_5_0(
+    V_9_9_0(
         "CorrelationCodec",
-        new Lucene95Codec(),
+        new Lucene99Codec(),
         new PerFieldCorrelationVectorsFormat(Optional.empty()),
         (userCodec, mapperService) -> new CorrelationCodec(userCodec, new PerFieldCorrelationVectorsFormat(Optional.of(mapperService))),
         CorrelationCodec::new
     );
 
-    private static final CorrelationCodecVersion CURRENT = V_9_5_0;
+    private static final CorrelationCodecVersion CURRENT = V_9_9_0;
```

Comment:
I don't think we can just replace this and be able to support in-place upgrades.

People who have already written indices with the 9.5 version of the CorrelationCodec will need to be able to read their old segments. With this change, I believe Lucene will say "CorrelationCodec 9.5? Never 'eard of 'im".

I think we'll need to create a separate correlation990 directory, and add another version this enum.

Reply:
Thanks for calling this out, working on updating this with a bwc test as well for the plugin

Reply:
@msfroh Went down this path but it looks like this plugin was not released yet in 2.11 https://github.com/opensearch-project/OpenSearch/tree/2.11/plugins and is currently not in 2.x? https://github.com/opensearch-project/OpenSearch/tree/2.x/plugins cc @sbcd90 I see you worked on this plugin, are there plans to backport/release this?  If not I think we are safe to simply bump and not provide 9.5 for bwc?

Reply:
hi @mch2 , the plugin has not been launched yet. the plugin development is in progress. we have open pr s that need to be merged in (https://github.com/opensearch-project/OpenSearch/pull/7771 ) before we backport/release this.

Reply:
thanks @sbcd90, are we safe to introduce this plugin with 99 as default?

Reply:
hi @mch2 , we should be safe to introduce this plugin with 99 as default. the unit tests should verify if plugin behavior is correct with 99 as default.

Reply:
thanks @sbcd90, will bump this to 99 as the tests are passing.  Made an additional commit here to rename the package to `correlation990` from `correlation950`.